.changeset/proud-socks-slide.md

```diff
@@ -0,0 +1,19 @@
+---
+"@cloudflare/vite-plugin": patch
+---
+
+Fix: Ensure that `vite dev` and `vite preview` crash with an appropriate error message when a remote proxy session is required but it the connection with it fails to be established
```

Comment:
nit: I liked how @penalosa put it: "hard error" instead of "crash"
```suggestion
Fix: Ensure that `vite dev` and `vite preview` hard error with an appropriate error message when a remote proxy session is required but it the connection with it fails to be established
```